static/js/publisher/builds/components/repoConnect.js

```diff
@@ -0,0 +1,143 @@
+import React from "react";
+import ReactDOM from "react-dom";
+import PropTypes from "prop-types";
+
+import Select from "@canonical/react-components/dist/components/Select";
+import Button from "@canonical/react-components/dist/components/Button";
+import SearchSelect from "./searchSelect";
+
+class RepoConnect extends React.Component {
+  constructor(props) {
+    super(props);
+
+    // Add disabled "Select organization" item to the list
+    const organizations = [
+      { value: "", disabled: "disabled", label: "Select organization" },
+      ...this.props.values
+    ];
+
+    this.state = {
+      isReadyToBuild: false,
+      value: [{ key: "Glide", name: "3dfx Glide License" }],
+      selectedRepo: [],
+      repoList: [
+        { key: "Glide", name: "3dfx Glide License" },
+        { key: "Abstyles", name: "Abstyles License" },
+        { key: "AFL-1.1", name: "Academic Free License v1.1" },
+        { key: "AFL-1.2", name: "Academic Free License v1.2" },
+        { key: "AFL-2.0", name: "Academic Free License v2.0" },
+        { key: "AFL-2.1", name: "Academic Free License v2.1" },
+        { key: "AFL-3.0", name: "Academic Free License v3.0" },
+        {
+          key: "AMPAS",
+          name: "Academy of Motion Picture Arts and Sciences BSD"
+        },
+        { key: "APL-1.0", name: "Adaptive Public License 1.0" },
+        { key: "Adobe-Glyph", name: "Adobe Glyph List License" }
+      ],
+      organizations: organizations
+    };
+
+    this.updateRepoList = this.updateRepoList.bind(this);
+    this.updateSelectedRepo = this.updateSelectedRepo.bind(this);
+  }
+
+  /**
+   * Update the original input value and dispatch a change event to the input and form.
+   */
+  // updateHandler(input, delimiter) {
+  //   const _input = input;
+  //   const _delimiter = delimiter;
+  //   return function(values) {
+  //     _input.value = values.map(item => item.key).join(_delimiter);
+  //     const changeEvent = new Event("change", { bubbles: true });
+
+  //     _input.dispatchEvent(changeEvent);
+  //   };
+  // }
+
+  /**
+   * Handle repo select
+   * *
+   * @param organization
+   */
+  handleRepoSelect(selectedRepo) {
+    if (selectedRepo.length === 0) {
+      this.setState({ isReadyToBuild: false });
+    } else {
+      this.setState({ isReadyToBuild: true });
+    }
+  }
+
+  /**
+   * Update selected repo from the searchSelect component
+   *
+   * @param organization
+   */
+  updateSelectedRepo(selectedRepo) {
+    this.setState({ selectedRepo: selectedRepo }, () => {
+      this.handleRepoSelect(this.state.selectedRepo);
+    });
+  }
+
+  /**
+   * Update repo list with repos of the selected organization
+   *
+   * @param value
+   */
+  updateRepoList(value) {
+    const organization = this.state.organizations.filter(
```

Comment:
I think the value and label for repo's should be the same - whatever is shown in the url bar.
That way when you read the row it'll say "canonical-web-and-design/snapcraft.io" not "Canonical web and design/snapcraft.io". It also means this filter won't be required.